src/lib/containers/access_requirement_list/managedACTAccess/CancelRequestDataAccess.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react'
+import { RequestInterface } from '../../../utils/synapseTypes'
+import * as ReactBootstrap from 'react-bootstrap'
+import { Alert, Button } from 'react-bootstrap'
+import { useState } from 'react'
+import { updateDataAccessRequest } from '../../../utils/SynapseClient'
+import { AlertProps } from './RequestDataAccessStep2'
+
+export type CancelRequestDataAccessProps = {
+  formSubmitRequestObject: RequestInterface | undefined,
+  token: string,
+  onHide: Function
+}
+
+const CancelRequestDataAccess:React.FC<CancelRequestDataAccessProps> = props => {
+
+  const {formSubmitRequestObject, token, onHide} = props
+  const [alert, setAlert] = useState<AlertProps | undefined>()
+  const [showCloseBtn, setShowCloseBtn] = useState<boolean>(false)
+
+  const handleSave = async () => {
+    if (formSubmitRequestObject) {
+      try {
+        const resp = await updateDataAccessRequest(formSubmitRequestObject, token)
+        if (resp.etag) {  // save success, close dialog
```

Comment:
Did you run into a case where the backend responded with a 200 but left etag undefined?  I did not expect to see this case (I would have thought that your catch would have been sufficient).

Reply:
I haven't seen it yet. What would be a better way to check if saving is successful if etag may not be reliable?

Reply:
The http response code should indicate the if the save was successful.  If the RPC was successful (200 level response) then we can assume there were no errors.  SynapseClient will only give you the response if it was successful.